mergify_engine/clients/github_app.py

```diff
@@ -0,0 +1,85 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2020 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import time
+
+import httpx
+import jwt
+
+from mergify_engine import config
+from mergify_engine.clients import common
+
+
+DEFAULT_ACCEPT = "application/vnd.github.machine-man-preview+json"
+
+
+class GithubBearerAuth(httpx.Auth):
+    JWT_EXPIRATION = 60
+
+    def __init__(self):
+        self.jwt = None
+        self.jwt_expiration = None
+
+    def get_or_create_jwt(self):
+        now = int(time.time())
+
+        if self.jwt is None or self.jwt_expiration <= now:
+            self.jwt_expiration = now + self.JWT_EXPIRATION
+            payload = {
+                "iat": now,
+                "exp": self.jwt_expiration,
+                "iss": config.INTEGRATION_ID,
+            }
+            encrypted = jwt.encode(payload, key=config.PRIVATE_KEY, algorithm="RS256")
+            self.jwt = encrypted.decode("utf-8")
+
+        return self.jwt
+
+    def auth_flow(self, request):
+        request.headers["Authorization"] = f"Bearer {self.get_or_create_jwt()}"
+        response = yield request
+        if response.status_code == 401:
+            self.jwt = None
+            request.headers["Authorization"] = f"Bearer {self.get_or_create_jwt()}"
+            yield request
+
+
+class _Client(httpx.Client, common.HttpxRetriesMixin):
+    def __init__(self):
+        super().__init__(
+            base_url=f"https://api.{config.GITHUB_DOMAIN}",
+            headers={"Accept": DEFAULT_ACCEPT, "User-Agent": "Mergify/Python"},
```

Comment:
maybe don't repeat this string for user agent?

Reply:
Done